dataset_conversion/create_data_splits.py

```diff
@@ -0,0 +1,125 @@
+"""
+Creates train/validation/test splits for a given dataset and saves the output in JSON file. 
+The JSON file can be used to convert the dataset into nnUNet format using the script convert_bids_to_nnunet.py
+
+Usage: 
+1. If you want to include specific sessions and contrasts:
```

Comment:
I also tested this case only for `ses-02` on `sci-zurich` using:

```
python dataset_conversion/create_data_splits.py --path-data < PATH_TO_DATASET >/sci-zurich --path-out ./ --split 0.8 0.1 0.1 --include-contrasts acq-sag_T2w --include-sessions ses-02
```

The output `split-dataset.json` file looks fine:

```json
{
    "numTest": 2,
    "numTraining": 5,
    "numValidation": 1,
    "seed": 42,
...
```

But, the script printed to CLI numbers for the whole dataset (i.e. 97 subjects):

```
2023-03-15 11:45:07.001 | INFO     | __main__:<module>:64 - Total number of subjects in the dataset: 97
Loading train volumes: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 77/77 [00:00<00:00, 8588.71it/s]
Loading valid volumes: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 8312.14it/s]
Loading test volumes: 100%|████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:00<00:00, 8149.03it/s]
```

I am just double-checking that this is the intention; the script loop across all subjects in the input dataset and includes only those with `ses-02`, right?

Reply:
Yes, exactly! The main loop is across all the subjects but in the json file only the subjects included by the user are shown. This is intended because one can easily look at the json at check how many subjects have been used for train/val/test

Reply:
Okay! Great, thanks for the explanation! Resolving the conversation.